File: backend/app/schemas/mortgage.py
from pydantic import BaseModel, Field, condecimal, PositiveInt, NonNegativeFloat

class MortgageRequest(BaseModel):
    principal: condecimal(gt=0) = Field(..., description="Loan amount")
    annual_interest_rate: NonNegativeFloat = Field(..., description="Annual % rate, e.g., 7.5")
    term_years: PositiveInt = Field(..., description="Loan term in years")
    extra_payment: NonNegativeFloat = 0.0
    compounding_per_year: PositiveInt = 12

class PaymentBreakdown(BaseModel):
    month: PositiveInt
    interest: float
    principal: float
    balance: float

class MortgageResponse(BaseModel):
    monthly_payment: float
    total_interest: float
    total_paid: float
    payoff_months: int
    schedule: list[PaymentBreakdown]

File: frontend/src/api/mortgage.ts
type Req = {
  principal: number
  annual_interest_rate: number
  term_years: number
  extra_payment?: number
  compounding_per_year?: number
}

export type Payment = { month: number; interest: number; principal: number; balance: number }
export type Resp = {
  monthly_payment: number
  total_interest: number
  total_paid: number
  payoff_months: number
  schedule: Payment[]
}

import { api } from './client'
export const calculateMortgage = async (payload: Req) => {
  const { data } = await api.post<Resp>('/api/mortgage/calculate', payload)
  return data
}
